File: zoomx/src/main/java/com/zoomx/zoomx/util/FormatUtil.java
package com.zoomx.zoomx.util;

import java.io.IOException;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

import okhttp3.RequestBody;
import okio.Buffer;

/**
 * Created by Ibrahim AbdelGawad on 12/12/2017.
 */

public class FormatUtil {

    final public static String DATE_FORMAT = "yyyy-MM-dd hh:mm:ss a";

    public static String formatDate(Date date, String format) {
        String formattedString = "";
        if (date != null) {
            try {
                SimpleDateFormat simpleDateFormat = new SimpleDateFormat(format);
                formattedString = simpleDateFormat.format(date);
            } catch (Exception e) {
                return "";
            }
        }
        return formattedString;
    }

    public static String bodyToString(final RequestBody request){
        try {
            final Buffer buffer = new Buffer();
            if(request != null)
                request.writeTo(buffer);
            else
                return "";
            return buffer.readUtf8();
        }
        catch (final IOException e) {
            return "did not work";
        }
    }

}

File: zoomx/src/main/java/com/zoomx/zoomx/util/FormatUtil.kt
package com.zoomx.zoomx.util

import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by Ibrahim AbdelGawad on 12/12/2017.
 */

object FormatUtil {

    const val DATE_FORMAT = "yyyy-MM-dd hh:mm:ss a"

    fun formatDate(date: Date?, format: String): String {
        var formattedString = ""
        if (date != null) {
            try {
                val simpleDateFormat = SimpleDateFormat(format, Locale.getDefault())
                formattedString = simpleDateFormat.format(date)
            } catch (e: Exception) {
                return ""
            }

        }
        return formattedString
    }

}
